### PR TITLE
Codecs.XorCodec

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/util/dynamic/Codecs.mapping [hunk 1]
```diff
@@ -26,10 +26,52 @@ CLASS net/minecraft/class_5699 net/minecraft/util/dynamic/Codecs
...
+		ARG 1 second
+			COMMENT the second codec
 	CLASS 1
 		METHOD read (Lcom/mojang/serialization/DynamicOps;Ljava/lang/Object;)Lcom/mojang/serialization/DataResult;
 			ARG 1 ops
 			ARG 2 input
 		METHOD write (Lcom/mojang/serialization/DynamicOps;Ljava/lang/Object;)Ljava/lang/Object;
 			ARG 1 ops
 			ARG 2 value
+	CLASS class_5839 XorCodec
```
Comment (i509VCB) [inline mappings/net/minecraft/util/dynamic/Codecs.mapping]:
```suggestion
	CLASS class_5839 Xor
```

Codec suffix is redundant in a class named `Codecs`

**Review by i509VCB — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/dynamic/Codecs.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/util/dynamic/Codecs.mapping]: mojang tostring says `XorCodec`

Comment (i509VCB) [reply mappings/net/minecraft/util/dynamic/Codecs.mapping]: I think that's a pretty weak justification, also consider GameRules does not suffix the inner classes except for the rule subclasses I believe.

**Comment by liach** [comment]
Ok how can I request a review from @i509vcb again

**Comment by liach** [comment]
Frankly this is one of the better parts of today's snapshot. The improved censorship system is so bad

**Comment by i509VCB** [comment]
> The improved censorship system is so bad

Regex hell?

**Review by i509VCB — APPROVED** [review]

File: mappings/net/minecraft/util/dynamic/Codecs.mapping [hunk 2]
```diff
@@ -26,10 +26,52 @@ CLASS net/minecraft/class_5699 net/minecraft/util/dynamic/Codecs
 		ARG 1 element
 	METHOD method_32852 (Lcom/mojang/serialization/DynamicOps;Ljava/lang/Object;)Z
 		ARG 1 element
+	METHOD method_33817 xor (Lcom/mojang/serialization/Codec;Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
+		COMMENT Returns a exclusive-or codec for {@link Either} instances.
```
Comment (YanisBft) [inline mappings/net/minecraft/util/dynamic/Codecs.mapping]:
```suggestion
		COMMENT Returns an exclusive-or codec for {@link Either} instances.
```

**Review by YanisBft — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**